src/action/base/create_or_merge_nix_config.rs

```diff
@@ -0,0 +1,505 @@
+                        // check.
+                    } else if MERGEABLE_CONF_NAMES.contains(&pending_conf_name.as_str()) {
+                        let pending_conf_value = pending_conf_value.join(" ");
+                        let existing_conf_value = existing_conf_value.join(" ");
+
+                        merged_nix_config.insert(
+                            pending_conf_name.to_owned(),
+                            format!("{pending_conf_value} {existing_conf_value}"),
+                        );
+                    } else {
+                        unmergeable_config_names.push(pending_conf_name.to_owned());
+                    }
+                } else {
+                    merged_nix_config
+                        .insert(pending_conf_name.to_owned(), pending_conf_value.to_owned());
+                }
+            }
+
+            if !unmergeable_config_names.is_empty() {
+                return Err(ActionError::UnmergeableConfig(
+                    unmergeable_config_names,
+                    this.path,
+                ));
+            }
+
+            if !merged_nix_config.is_empty() {
+                this.nix_configs.merged_nix_config = Some(merged_nix_config);
+                return Ok(StatefulAction::uncompleted(this));
+            }
+
+            tracing::debug!(
+                "File `{}` already contains what we want",
+                this.path.display()
+            );
+            return Ok(StatefulAction::completed(this));
+        } else {
+            let mut merged_nix_config: NixConfig = NixConfig::new();
+            for (pending_conf_name, pending_conf_value) in &this.nix_configs.pending_nix_config {
+                merged_nix_config
+                    .insert(pending_conf_name.to_owned(), pending_conf_value.to_owned());
+            }
+
+            if !merged_nix_config.is_empty() {
+                this.nix_configs.merged_nix_config = Some(merged_nix_config);
+            }
+        }
+
+        Ok(StatefulAction::uncompleted(this))
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "create_or_merge_nix_config")]
+impl Action for CreateOrMergeNixConfig {
+    fn tracing_synopsis(&self) -> String {
+        format!("Create or overwrite file `{}`", self.path.display())
+    }
+
+    fn tracing_span(&self) -> Span {
+        let span = span!(
+            tracing::Level::DEBUG,
+            "create_file",
+            path = tracing::field::display(self.path.display()),
+            user = self.user,
+            group = self.group,
+            mode = self
+                .mode
+                .map(|v| tracing::field::display(format!("{:#o}", v))),
+            buf = tracing::field::Empty,
+        );
+
+        if tracing::enabled!(tracing::Level::TRACE) {
+            span.record("buf", &self.buf);
+        }
+        span
+    }
+
+    fn execute_description(&self) -> Vec<ActionDescription> {
+        vec![ActionDescription::new(self.tracing_synopsis(), vec![])]
```

Comment:
Since this is actually intelligently merging the config, we should tell them that in the `vec![]` so it would show up with `--explain`.